examples/question_answering/question_answering/workflow.py

```diff
@@ -0,0 +1,199 @@
+# Copyright 2021-2023 Kolena Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import dataclasses
+from typing import List
+
+from pydantic.dataclasses import dataclass
+
+from kolena.workflow import define_workflow
+from kolena.workflow import EvaluatorConfiguration
+from kolena.workflow import GroundTruth as BaseGroundTruth
+from kolena.workflow import Inference as BaseInference
+from kolena.workflow import Metadata
+from kolena.workflow import MetricsTestCase
+from kolena.workflow import MetricsTestSample
+from kolena.workflow import MetricsTestSuite
+from kolena.workflow import Text
+from kolena.workflow.annotation import ClassificationLabel
+
+
+@dataclass(frozen=True)
+class TestSample(Text):
+    """Test sample type for the Question Answering workflow."""
+
+    data_id: str
+    """The identifier of the story."""
+
+    question: str
+    """The question based on the story."""
+
+    turn: int
+    """The index of the question asked for the same story."""
+
+    metadata: Metadata = dataclasses.field(default_factory=dict)
+    """The metadata associated with the test sample."""
+
+
+@dataclass(frozen=True)
+class GroundTruth(BaseGroundTruth):
+    """Ground truth type for the Question Answering workflow."""
+
+    answer: str
+    """The human created answer for the question based on the story."""
+
+    clean_answer: str
+    """A normalized and cleaned copy of the answer"""
+
+    context: List[str]
+    """The questions and answers in the history of the conversation."""
+
+    other_answer_1: str
+    """An alternative answer for the question."""
+
+    other_answer_2: str
+    """An alternative answer for the question."""
+
+    other_answer_3: str
+    """An alternative answer for the question."""
+
+    question_answer: ClassificationLabel
+    """A label to display the question and answer with the story."""
+
+
+@dataclass(frozen=True)
+class Inference(BaseInference):
+    """Inference type for the Question Answering workflow."""
+
+    answer: str
```

Comment:
What do you think of using `ClassificationLabel` for `answer` to show the answer as an annotation?